c++/src/kj/compat/brotli.c++

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) 2023 Cloudflare, Inc. and contributors
+// Licensed under the MIT License:
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#if KJ_HAS_BROTLI
+
+#include "brotli.h"
+#include <kj/debug.h>
+
+namespace kj {
+
+namespace _ {  // private
+
+// Check number of window bits used by the stream, see RFC 7932 section 9.1 for the specification.
+// Adapted from FL's ngx_brotli.
+static int get_brotli_window_bits(uint8_t peek) {
```

Comment:
Non-blocking nit: we tend to prefer enclosing things in an anonymous namespace (`namespace { ... }`) instead of `static` qualifiers. I think they're exactly equivalent, though.